### PR TITLE
[Merged by Bors] - feat: port Order.CompleteLatticeIntervals

### PR DESCRIPTION
**Pull request by ChrisHughes24**

### CODE REVIEW

File: Mathlib/Order/CompleteLatticeIntervals.lean [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+theorem infₛ_within_of_ordConnected {s : Set α} [hs : OrdConnected s] ⦃t : Set s⦄ (ht : t.Nonempty)
+    (h_bdd : BddBelow t) : infₛ ((↑) '' t : Set α) ∈ s := by
+  obtain ⟨c, hct⟩ : ∃ c, c ∈ t := ht
+  obtain ⟨B, hB⟩ : ∃ B, B ∈ lowerBounds t := h_bdd
+  refine' hs.out B.2 c.2 ⟨_, _⟩
+  · exact (Subtype.mono_coe s).le_cinfₛ_image ⟨c, hct⟩ hB
+  · exact (Subtype.mono_coe s).cinfₛ_image_le hct ⟨B, hB⟩
+#align Inf_within_of_ord_connected infₛ_within_of_ordConnected
+
+/-- A nonempty `ord_connected` set in a conditionally complete linear order is naturally a
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/CompleteLatticeIntervals.lean:158]:
```suggestion
/-- A nonempty `OrdConnected` set in a conditionally complete linear order is naturally a
```

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Order/CompleteLatticeIntervals.lean [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+set_option linter.unusedVariables false in --Porting note: Variables are actually used - linter bug?
+/-- For a nonempty subset of a conditionally complete linear order to be a conditionally complete
+linear order, it suffices that it contain the `supₛ` of all its nonempty bounded-above subsets, and
+the `infₛ` of all its nonempty bounded-below subsets.
+See note [reducible non-instances]. -/
+@[reducible]
+noncomputable def subsetConditionallyCompleteLinearOrder [Inhabited s]
+    (h_Sup : ∀ {t : Set s} (_ : t.Nonempty) (h_bdd : BddAbove t), supₛ ((↑) '' t : Set α) ∈ s)
+    (h_Inf : ∀ {t : Set s} (_ : t.Nonempty) (h_bdd : BddBelow t), infₛ ((↑) '' t : Set α) ∈ s) :
```
Comment (riccardobrasca) [inline Mathlib/Order/CompleteLatticeIntervals.lean:110]:
```suggestion
/-- For a nonempty subset of a conditionally complete linear order to be a conditionally complete
linear order, it suffices that it contain the `supₛ` of all its nonempty bounded-above subsets, and
the `infₛ` of all its nonempty bounded-below subsets.
See note [reducible non-instances]. -/
@[reducible]
noncomputable def subsetConditionallyCompleteLinearOrder [Inhabited s]
    (h_Sup : ∀ {t : Set s} (_ : t.Nonempty) (_h_bdd : BddAbove t), supₛ ((↑) '' t : Set α) ∈ s)
    (h_Inf : ∀ {t : Set s} (_ : t.Nonempty) (_h_bdd : BddBelow t), infₛ ((↑) '' t : Set α) ∈ s) :
```
I am not sure this is the correct solution, but it makes the linter happy, and it is what Scott does in his video.

Comment (riccardobrasca) [reply Mathlib/Order/CompleteLatticeIntervals.lean:110]: In any case LGTM.

bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**